From 5e7a6cf48c2ca4826d96cba1026669ecf3893d66 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 28 May 2018 10:41:12 +0800
Subject: [PATCH 4513/5242] MLK-19413-2 arm64: fsl-imx8qm.dtsi: Add pixel
 combiner support

commit  d6c051475cc45b649d49de7171c1b913e6adeff2 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds pixel combiner nodes support for i.MX8qm DT
file and hooks the pixel combiner nodes to the DPU nodes.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 512c2fe..6cb4313 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -1537,6 +1537,13 @@
 		reg = <0x0 0x56000000 0x0 0x10000>;
 	};
 
+	pixel_combiner1: pixel-combiner@56020000 {
+		compatible = "fsl,imx8qm-pixel-combiner";
+		reg = <0x0 0x56020000 0x0 0x10000>;
+		power-domains = <&pd_dc0>;
+		status = "disabled";
+	};
+
 	prg1: prg@56040000 {
 		compatible = "fsl,imx8qm-prg";
 		reg = <0x0 0x56040000 0x0 0x10000>;
@@ -1744,6 +1751,7 @@
 		fsl,dpr-channels = <&dpr1_channel1>, <&dpr1_channel2>,
 				   <&dpr1_channel3>, <&dpr2_channel1>,
 				   <&dpr2_channel2>, <&dpr2_channel3>;
+		fsl,pixel-combiner = <&pixel_combiner1>;
 		status = "disabled";
 
 		dpu1_disp0: port@0 {
@@ -2045,6 +2053,13 @@
 		reg = <0x0 0x57000000 0x0 0x10000>;
 	};
 
+	pixel_combiner2: pixel-combiner@57020000 {
+		compatible = "fsl,imx8qm-pixel-combiner";
+		reg = <0x0 0x57020000 0x0 0x10000>;
+		power-domains = <&pd_dc1>;
+		status = "disabled";
+	};
+
 	prg10: prg@57040000 {
 		compatible = "fsl,imx8qm-prg";
 		reg = <0x0 0x57040000 0x0 0x10000>;
@@ -2252,6 +2267,7 @@
 		fsl,dpr-channels = <&dpr3_channel1>, <&dpr3_channel2>,
 				   <&dpr3_channel3>, <&dpr4_channel1>,
 				   <&dpr4_channel2>, <&dpr4_channel3>;
+		fsl,pixel-combiner = <&pixel_combiner2>;
 		status = "disabled";
 
 		dpu2_disp0: port@0 {
-- 
1.7.9.5

